src/mem/cache/prefetch/base.cc

```diff
@@ -238,6 +238,7 @@ Base::probeNotify(const CacheAccessProbeArg &acc, bool miss)
     // operations or for writes that we are coaslescing.
```

Comment:
Side comment - can be ignored: I would say that the new line should have been added here, but the problem is that this comment, as it is, does not add any relevant information. Ideally this comment should inform why the prefetcher is not notified in these scenarios. It would be nice if you could add this why info to the new line you are adding in this PR. It would be extra nice if you could improve the whole comment to explain why the prefetcher is not notified in each of these scenarios.

Reply:
Hi Daniel. That's a fair comment. It looks like this section has become a collection of edge cases that have caused problems in the past. 
I'm not personally aware of the exact reasons for most of these early returns. Perhaps we could leave improving this documentation for a future commit?

Reply:
Sure, no problem!